File: csci40/lec21/pow.cpp
#include <iostream>
using namespace std;

// returns m^n
double pow(double m, int n) {
  if (n == 0) {
    // base case: anything to the 0th power is 1
    return 1.0;
  } else {
    // recursive case: m^n = m * m^(n-1)
    return m * pow(m, n-1);
  }
}

int main() {
  cout << pow(2.5, 5) << endl;
  cout << pow(2, 4) << endl;
  cout << pow(2, 5) << endl;

  return 0;
}

File: csci1/lec30/pow.py
# return n**m
def pow(n, m):
    if m == 0:
        # anything to the 0th power is 1
        return 1
    else:
        # recursive case: n^m = n * n^(m-1)
        return n * pow(n, m-1)

print(pow(2, 4))
print(pow(2.5, 5))
